cmd/evergreen/evergreen.go

```diff
@@ -100,6 +102,19 @@ func buildApp() *cli.App {
 	}
 
 	app.Before = func(c *cli.Context) error {
+		confPath := c.String("conf")
+		conf, err := operations.NewClientSettings(confPath)
+		if err != nil {
+			return errors.Wrap(err, "problem loading configuration")
+		}
+		if conf.AutoUpgradeCli {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			err = operations.DoUpdate(conf, ctx, true, false)
```

Comment:
L113-115 could be simplified down to `return errors.Wrap(err, "problem auto-updating CLI")`.